### ISSUE TITLE
MQTT disconnects

### ISSUE THREAD

**Issue by MYeager1967** [VorlonCD/bi-aidetection#86]
Is the AITool supposed to disconnect from the broker after every message? Just curious to see if this is the expected behavior or if there's an issue (either with the software or possibly on my end)

**Comment by VorlonCD** [comment]
Yes, I would expect it to disconnect after every message the way it is currently set up.  How should it behave?  Does it cause any trouble by disconnecting every time?

Side note - noticed you posed on DOODS github project.   Trying to get that to work right now for AITOOL but its telling me the test image I am sending it is too large.   Granted, 6mb jpeg probably 4k but still...  why wouldn't it accept images of any size.   The doods page says it will auto resize if needed.

**Comment by MYeager1967** [comment]
I'm going to assume your response was to two different people. Most MQTT clients I believe stay connected as best they can. It's not causing any issues, I just wondered if that was how it was supposed to act. As for DOODS, I don't believe I posted on any DOODS topics in quite a while so I think that was for someone else.

DOODS is a different animal and it works well. A version to support DOODS would benefit some people. Hell, I might even give it a look.... 😁

**Comment by VorlonCD** [comment]
Now how could I miss that little picture of a clown that in no way haunts me?  :)
https://github.com/snowzach/doods/issues/29

**Comment by MYeager1967** [comment]
That was from a while back. I thought you meant currently. Going back and reading your earlier message, I understand it better now. Anyhow, I literally sent DOODS the hires image straight from my 4k cameras. Did it through an http request though. Seems AITool uses the same process so it should be good. I used DOODS for a while, worked really good with the right models file. Tried Frigate, not enough hardware muscle at the time. Even played with the  tensorflow setup for HA. Quickly got away from anything too integrated into HA as it was just another thing to break.

**Comment by Kosh42** [comment]
Rather than me create a new issue, can this one be re-opened? I have multiple cameras and use a short cool down (AI is on a meaty machine), so my MQTT server is being fairly spammed by AITool connect and disconnects, and it's fairly well used as well. I think this may be what is slowing down my HA instance. I'm going to re-base this into the same server eventually when everything else is stable, but if possible would like to see AITool connect to the server, then just post it's topic updates as each image is processed/the cancel message on cooldown like every other MQTT client does?

And many thanks to all for your great work on this. I need to learn code so I can be part of the answer rtaher than adding to work loads, I know.

**Comment by edalquist** [comment]
+1 to keep the MQTT connection open, it is fairly cheap to setup but still costs time and CPU on both ends to reconnect.

**Comment by VorlonCD** [comment]
@Kosh42 - (Vorlon!), try this.   Consider it very beta since I just changed code without actually testing with real mqtt server - backup the whole AITOOL folder first so you can revert back if needed.   Unblock the downloaded zip and overwrite everything.

It now creates a connection to the server once per session.  If it looses connection, in theory it should re-connect, but not every time a message is sent.

I'll need logs if doesnt seem to be working as expected.

[AITOOL-VORLONCD.zip](https://github.com/VorlonCD/bi-aidetection/files/5743716/AITOOL-VORLONCD.zip)

**Comment by Kosh42** [comment]
Many thanks for the speedy reply. Installed and all seems to be working fine. It's actually massively speeded up my average detection times as well, even though I was passing the MQTT post over to another thread before.

In all seriousness, you need a "buy me a beer" link for people who'd like to say thank you for your work on this.

**Comment by VorlonCD** [comment]
Ok great!  I assume it speed things up because the MQTT call doesnt take as long?    You could have also speed it up by enable camera > actions > queue actions.   The action is sent to a background thread so the main URL detect objects thread is freed up immediately for the next image.

**Comment by Kosh42** [comment]
I had already passing it off to queue on another thread. But now it's not connecting them disconnecting MQTT messages every image and just publishing, average time on AITool is quicker, and my MQTT broker is no longer lagging.

Thank you again.

**Comment by edalquist** [comment]
My AITool and MQTT server are much happier with the constant connection.

**Comment by VorlonCD** [comment]
Ok, great!!

**Comment by MYeager1967** [comment]
Vorlon, your commitment to this utility is wonderful. It just keeps getting better!!!
